google-cloud-datastore/src/main/java/com/google/cloud/datastore/execution/response/AggregationQueryResponseTransformer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.datastore.execution.response;
+
+import com.google.api.core.InternalApi;
+import com.google.cloud.Timestamp;
+import com.google.cloud.datastore.AggregationResult;
+import com.google.cloud.datastore.AggregationResults;
+import com.google.cloud.datastore.LongValue;
+import com.google.datastore.v1.RunAggregationQueryResponse;
+import com.google.datastore.v1.Value;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@InternalApi
+public class AggregationQueryResponseTransformer
+    implements ResponseTransformer<RunAggregationQueryResponse, AggregationResults> {
+
+  @Override
+  public AggregationResults transform(RunAggregationQueryResponse response) {
+    Timestamp readTime = Timestamp.fromProto(response.getBatch().getReadTime());
+    List<AggregationResult> aggregationResults =
+        response.getBatch().getAggregationResultsList().stream()
+            .map(
+                aggregationResult -> new AggregationResult(resultWithLongValues(aggregationResult)))
+            .collect(Collectors.toCollection(LinkedList::new));
```

Comment:
why a linked list here?

Reply:
while writing this i felt LinkedList will be more performant here because:
 - If we have too many `AggregationResult` coming in, then `ArrayList` will need to double it's capacity everytime it's full and copy the existing data in it whereas in `LinkedList` it'll be a O(1) operation.
 - We are exposing the result as an iterator anyway and won't need a case where we need to access the result using random index where `ArrayList` is optimized.